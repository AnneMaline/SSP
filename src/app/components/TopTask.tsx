// src/components/Button.tsx
"use client";
import Link from "next/link";
import { useSearchParams } from "next/navigation";
import React from "react";

interface TopTaskProps {
  text: string; // Text on the top task
  targetUrl: string; // target URL for navigation
  className?: string; // Optional className prop for customization
}

const TopTask: React.FC<TopTaskProps> = ({
  text,
  targetUrl,
  className = "",
}) => {
  const searchParams = useSearchParams(); // Access the current search parameters
  return (
    <Link
      href={`${targetUrl}?${searchParams}`}
      className={`px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 ${className}`}
    >
      {text}
    </Link>
  );
};

export default TopTask;
