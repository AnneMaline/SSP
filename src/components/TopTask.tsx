import Link from "next/link";
import React from "react";

interface TopTaskProps {
  text: string; // Text on the top task
  targetUrl: string; // target URL for navigation
  className?: string; // Optional className prop for customization
}

const TopTask = ({ text, targetUrl, className = "" }: TopTaskProps) => {
  return (
    <Link
      href={`${targetUrl}`}
      className={`px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 ${className}`}
    >
      {text}
    </Link>
  );
};

export default TopTask;
