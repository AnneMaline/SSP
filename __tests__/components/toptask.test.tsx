import { render, screen } from "@testing-library/react";
import TopTask from "../../src/app/components/TopTask";
import { ReadonlyURLSearchParams, useSearchParams } from "next/navigation";
import { mocked } from "jest-mock";

// Mock useSearchParams
jest.mock("next/navigation", () => ({
  useSearchParams: jest.fn(),
}));

describe("TopTask", () => {
  const mockUseSearchParams = mocked(useSearchParams);

  beforeEach(() => {
    // Mock default behavior of useSearchParams
    const mockSearchParams: Partial<ReadonlyURLSearchParams> = {
      toString: () => "param1=value1&param2=value2",
      get: (name: string) => {
        const params: { [key: string]: string } = {
          param1: "value1",
          param2: "value2",
        };
        return params[name] || null;
      },
      getAll: () => [],
      has: () => false,
      keys: jest.fn().mockReturnValue([][Symbol.iterator]()),
      values: jest.fn().mockReturnValue([][Symbol.iterator]()),
      entries: jest.fn().mockReturnValue([][Symbol.iterator]()),
      forEach: jest.fn(),
    };

    // Cast the mock as ReadonlyURLSearchParams
    mockUseSearchParams.mockReturnValue(
      mockSearchParams as ReadonlyURLSearchParams
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const { container } = render(
      <TopTask
        text="Go to Target"
        targetUrl="/target-page"
        className="custom-class"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the link with the correct text and href", () => {
    render(
      <TopTask
        text="Go to Target"
        targetUrl="/target-page"
        className="custom-class"
      />
    );

    // Check if the link text is correct
    const link = screen.getByRole("link", { name: /go to target/i });
    expect(link).toBeInTheDocument();

    // Check if the href is correct
    expect(link).toHaveAttribute(
      "href",
      "/target-page?param1=value1&param2=value2"
    );

    // Check if custom className is applied
    expect(link).toHaveClass("custom-class");
  });

  it("updates href dynamically based on search parameters", () => {
    render(<TopTask text="Dynamic Params" targetUrl="/dynamic-page" />);

    const link = screen.getByRole("link", { name: /dynamic params/i });

    // Check updated href
    expect(link).toHaveAttribute(
      "href",
      "/dynamic-page?param1=value1&param2=value2"
    );
  });
});
