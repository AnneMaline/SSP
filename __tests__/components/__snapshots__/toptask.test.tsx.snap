// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTask renders correctly 1`] = `
<div>
  <a
    class="px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 custom-class"
    href="/target-page?param1=value1&param2=value2"
  >
    Go to Target
  </a>
</div>
`;
